finagle-thrift/src/main/ruby/lib/finagle-thrift/trace.rb

```diff
@@ -57,6 +57,10 @@ def tracer=(tracer)
     @tracer = tracer
   end
 
+  def stack=(stack)
```

Comment:
I see your point, definitely this issue could happen.  A few possible approaches here:
1.  Thread local - the architecture would require requests to be handled in T1 and if it requires additional work, it would pass the Trace data necessary to T2. 
2. Global - guarded with mutexes synchronized code.  This approach seems like a bit of a band aid and not solving the real problem.

What do you think?

I will definitely add this to the readme.